### PR TITLE
pldm: Catch exceptions precisely and printing it

### PR DESCRIPTION
**Pull request by Patel-Kamalkumar**
Correcting catch block in PLDM repo to print all exception precisely so pldm trace can be more
useful to identify defect easily.

### CODE REVIEW

File: oem/ibm/configurations/pdr/ibm,rainier-1s4u/11.json [hunk 1]
```diff
@@ -614,6 +614,44 @@
                         }
                     ]
                 },
+                {
```
Comment (Pavithrab7) [inline oem/ibm/configurations/pdr/ibm,rainier-1s4u/11.json:617]: why do you have these changes in the PR?

File: oem/ibm/configurations/pdr/ibm,rainier-1s4u/4.json [hunk 2]
```diff
@@ -639,6 +639,44 @@
                         }
                     ]
                 },
+                {
```
Comment (Pavithrab7) [inline oem/ibm/configurations/pdr/ibm,rainier-1s4u/4.json:642]: same here as above, why these changes?

File: oem/ibm/libpldmresponder/inband_code_update.cpp [hunk 3]
```diff
@@ -335,15 +338,15 @@ void CodeUpdate::setVersions()
...
                 try
                 {
-                    auto propVal =
-                        pldm::utils::DBusHandler().getDbusPropertyVariant(
-                            imageObjPath, "Activation", imageInterface);
-                    const auto& activation = std::get<std::string>(propVal);
-
-                    if (activation ==
-                        "xyz.openbmc_project.Software.Activation.Activations.Invalid")
+                    if (isCodeUpdateInProgress())
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/inband_code_update.cpp:341]: why this change?

File: oem/ibm/libpldmresponder/inband_code_update.cpp [hunk 4]
```diff
@@ -106,6 +106,9 @@ int CodeUpdate::setNextBootSide(const std::string& nextSide)
     catch (const std::exception& e)
     {
         // Alternate side may not be present due to a failed code update
+        std::cerr
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/inband_code_update.cpp:109]: lg2 style of logging?

File: oem/ibm/libpldmresponder/file_io_type_lid.hpp [hunk 5]
```diff
@@ -21,9 +24,13 @@ using MarkerLIDremainingSize = uint64_t;
...
 constexpr auto accessKeyExpired =
-    "sdbusplus::xyz::openbmc_project::Software::Version::Error::ExpiredAccessKey";
+    "xyz.openbmc_project.Software.Version.Error.ExpiredAccessKey";
+using AccessKeyExpired =
+    sdbusplus::xyz::openbmc_project::Software::Version::Error::ExpiredAccessKey;
 constexpr auto incompatibleErr =
-    "sdbusplus::xyz::openbmc_project::Software::Version::Error::Incompatible";
+    "xyz.openbmc_project.Software.Version.Error.Incompatible";
+using IncompatibleErr =
+    sdbusplus::xyz::openbmc_project::Software::Version::Error::Incompatible;
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/file_io_type_lid.hpp:33]: why these changes in this PR?

File: oem/ibm/libpldmresponder/file_io.cpp [hunk 6]
```diff
@@ -831,6 +834,8 @@ Response Handler::fileAck(const pldm_msg* request, size_t payloadLength)
 
     catch (const InternalFailure& e)
     {
+        std::cerr << "unknown file type, TYPE=" << fileType
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/file_io.cpp:837]: lg2 style of logging

File: oem/ibm/libpldmresponder/collect_slot_vpd.cpp [hunk 7]
```diff
@@ -76,7 +76,7 @@ void SlotHandler::processSlotOperations(const std::string& slotObjectPath,
         // get the adapter dbus object path from the slot dbus object path
         adapterObjPath = getAdapterObjPath(slotObjectPath).value();
     }
-    catch (const std::bad_optional_access& e)
+    catch (const std::bad_optional_access& /*e*/)
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/collect_slot_vpd.cpp:79]: do we need a error message here?

File: oem/ibm/libpldmresponder/utils.cpp [hunk 8]
```diff
@@ -496,7 +496,10 @@ bool checkFruPresence(const char* objPath)
         isPresent = std::get<bool>(propVal);
     }
     catch (const sdbusplus::exception::SdBusError& e)
-    {}
+    {
+        std::cerr << "Failed to call the D-Bus Method"
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/utils.cpp:500]: lg2 style of logging

File: oem/ibm/configurations/pdr/ibm,rainier-1s4u/11.json (hunk 1 above)
Comment (Patel-Kamalkumar) [reply oem/ibm/configurations/pdr/ibm,rainier-1s4u/11.json]: corrected.

File: oem/ibm/libpldmresponder/file_io.cpp (hunk 6 above)
Comment (Patel-Kamalkumar) [reply oem/ibm/libpldmresponder/file_io.cpp]: done

File: oem/ibm/libpldmresponder/inband_code_update.cpp (hunk 4 above)
Comment (Patel-Kamalkumar) [reply oem/ibm/libpldmresponder/inband_code_update.cpp]: corrected.

File: oem/ibm/configurations/pdr/ibm,rainier-1s4u/4.json (hunk 2 above)
Comment (Patel-Kamalkumar) [reply oem/ibm/configurations/pdr/ibm,rainier-1s4u/4.json]: corrected

File: oem/ibm/libpldmresponder/file_io_type_lid.hpp (hunk 5 above)
Comment (Patel-Kamalkumar) [reply oem/ibm/libpldmresponder/file_io_type_lid.hpp]: corrected

File: oem/ibm/libpldmresponder/oem_ibm_handler.cpp [hunk 9]
```diff
@@ -1256,7 +1257,7 @@ bool pldm::responder::oem_ibm_platform::Handler::watchDogRunning()
         isWatchDogRunning = pldm::utils::DBusHandler().getDbusProperty<bool>(
             watchDogObjectPath, watchDogEnablePropName, watchDogInterface);
     }
-    catch (const std::exception& e)
+    catch (const std::exception& /*e*/)
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/oem_ibm_handler.cpp:1260]: error message here?

File: pldmtool/oem/ibm/pldm_oem_ibm.cpp [hunk 10]
```diff
@@ -148,8 +148,8 @@ class GetFileTable : public CommandInterface
 
         if (rc != PLDM_SUCCESS || cc != PLDM_SUCCESS)
         {
-            std::cerr << "Response Message Error: "
-                      << ", rc=" << rc << ", cc=" << (int)cc << std::endl;
+            lg2::error("Response Message Error: rc={RC} cc={CC}", "RC", rc,
```
Comment (Pavithrab7) [inline pldmtool/oem/ibm/pldm_oem_ibm.cpp:151]: why lg2::error ? just error()?

File: pldmtool/oem/ibm/pldm_oem_ibm.cpp [hunk 11]
```diff
@@ -71,8 +71,8 @@ class GetAlertStatus : public CommandInterface
 
         if (rc != PLDM_SUCCESS || completionCode != PLDM_SUCCESS)
         {
-            std::cerr << "Response Message Error: "
-                      << "rc=" << rc << ",cc=" << (int)completionCode << "\n";
+            lg2::error("Response Message Error: rc={RC} cc={CC}", "RC", rc,
```
Comment (Pavithrab7) [inline pldmtool/oem/ibm/pldm_oem_ibm.cpp:74]: why lg2::error()? just error()?

File: pldmtool/pldm_platform_cmd.cpp [hunk 12]
```diff
@@ -1397,10 +1397,8 @@ class GetNumericEffecterValue : public CommandInterface
 
         if (rc != PLDM_SUCCESS || completionCode != PLDM_SUCCESS)
         {
-            std::cerr << "Response Message Error: "
-                      << "rc=" << rc
-                      << ",cc=" << static_cast<int>(completionCode)
-                      << std::endl;
+            lg2::error("Response Message Error: rc={RC} cc={CC}", "RC", rc,
```
Comment (Pavithrab7) [inline pldmtool/pldm_platform_cmd.cpp:1400]: can we have the namespace added and remove lg2::error() and keep error() for uniformity?

File: requester/mctp_endpoint_discovery.cpp [hunk 13]
```diff
@@ -35,6 +35,7 @@ MctpDiscovery::MctpDiscovery(sdbusplus::bus_t& bus,
     }
     catch (const std::exception& e)
     {
+        error("Failed to call the D-Bus Method ERROR={ERR}", "ERR", e.what());
```
Comment (Pavithrab7) [inline requester/mctp_endpoint_discovery.cpp:38]: do we want to mention what DBus method failed?

Comment (Patel-Kamalkumar) [reply requester/mctp_endpoint_discovery.cpp:38]: yes, in exception case, we will not be able to identify where it's failed so trace is required.

Comment (Pavithrab7) [reply requester/mctp_endpoint_discovery.cpp:38]: yes, I got it but do you want to mention in the error the name of the DBus method?

File: oem/ibm/libpldmresponder/oem_ibm_handler.cpp (hunk 9 above)
Comment (Patel-Kamalkumar) [reply oem/ibm/libpldmresponder/oem_ibm_handler.cpp:1260]: not required error trace because it's checking watchdog is running or not, if not running then will return false.

File: requester/mctp_endpoint_discovery.cpp (hunk 13 above)
Comment (Patel-Kamalkumar) [reply requester/mctp_endpoint_discovery.cpp:38]: other Dbus exception catching block, method name is not mentioned, so keeping same as other traces.

File: pldmtool/oem/ibm/pldm_oem_ibm.cpp (hunk 10 above)
Comment (Patel-Kamalkumar) [reply pldmtool/oem/ibm/pldm_oem_ibm.cpp]: removed lg2:: tag

File: pldmtool/oem/ibm/pldm_oem_ibm.cpp (hunk 11 above)
Comment (Patel-Kamalkumar) [reply pldmtool/oem/ibm/pldm_oem_ibm.cpp]: removed lg2:: tag

File: pldmtool/pldm_platform_cmd.cpp (hunk 12 above)
Comment (Patel-Kamalkumar) [reply pldmtool/pldm_platform_cmd.cpp]: removed lg2:: tag

File: oem/ibm/libpldmresponder/utils.cpp [hunk 14]
```diff
@@ -494,7 +494,9 @@ bool checkFruPresence(const char* objPath)
         isPresent = std::get<bool>(propVal);
     }
     catch (const sdbusplus::exception::SdBusError& e)
-    {}
+    {
+        error("Failed to call the D-Bus Method ERROR={ERR}", "ERR", e.what());
```
Comment (jaypadath) [inline oem/ibm/libpldmresponder/utils.cpp:498]: Can you change something like "D-Bus method call to get the FRU presence failed" ?

File: oem/ibm/libpldmresponder/utils.cpp [hunk 15]
```diff
@@ -593,8 +596,9 @@ void findSlotObjects(const std::string& boardObjPath,
     }
     catch (const std::exception& e)
     {
-        error("no cec slots under motherboard {BOARD_OBJ_PATH}",
-              "BOARD_OBJ_PATH", boardObjPath.c_str());
+        error(
+            "no cec slots under motherboard {BOARD_OBJ_PATH} ERROR={ERR_EXCEP}",
```
Comment (jaypadath) [inline oem/ibm/libpldmresponder/utils.cpp:600]: Please change "no cec slots under" to "No cec slots found under"

File: requester/mctp_endpoint_discovery.cpp (hunk 13 above)
Comment (jaypadath) [reply requester/mctp_endpoint_discovery.cpp:38]: Kamal, rather than putting a generic D-bus error, can you mention something like "D-bus method call to get the managed objects under MCTP Control failed".

File: host-bmc/host_pdr_handler.cpp [hunk 16]
```diff
@@ -726,7 +724,7 @@ void HostPDRHandler::parseStateSensorPDRs()
         }
         // If there is no mapping for terminusHandle assign the reserved TID
         // value of 0xFF to indicate that.
-        catch (const std::out_of_range& e)
+        catch (const std::out_of_range& /*e*/)
```
Comment (jaypadath) [inline host-bmc/host_pdr_handler.cpp:727]: Please remove /*e*/. Its applicable to all the below files where you have done in similar way.

**Review by jaypadath — CHANGES_REQUESTED** [review]
In the commit message change "Catching exception precisely and printing it." to "pldm: Catch exceptions precisely and printing it".

Also add the Tested field in commit message.

File: requester/mctp_endpoint_discovery.cpp (hunk 13 above)
Comment (Patel-Kamalkumar) [reply requester/mctp_endpoint_discovery.cpp]: updated as per comment.

File: oem/ibm/libpldmresponder/utils.cpp (hunk 14 above)
Comment (Patel-Kamalkumar) [reply oem/ibm/libpldmresponder/utils.cpp]: updated as per comment.

File: oem/ibm/libpldmresponder/utils.cpp (hunk 15 above)
Comment (Patel-Kamalkumar) [reply oem/ibm/libpldmresponder/utils.cpp]: updated as per comment.

File: host-bmc/host_pdr_handler.cpp (hunk 16 above)
Comment (Patel-Kamalkumar) [reply host-bmc/host_pdr_handler.cpp]: updated as per comment.

File: oem/ibm/libpldmresponder/oem_ibm_handler.cpp [hunk 17]
```diff
@@ -1410,7 +1412,10 @@ void pldm::responder::oem_ibm_platform::Handler::handleBootTypesAtPowerOn()
             setBootTypesBiosAttr(restartCause);
         }
         catch (const std::exception& e)
-        {}
+        {
+            error("Failed to call the D-Bus Method ERROR={ERR}", "ERR",
```
Comment (jaypadath) [inline oem/ibm/libpldmresponder/oem_ibm_handler.cpp:1416]: Pls change "Failed to call the D-Bus Method" to "Failed to set the D-bus property for the Host restart reason"

**Review by jaypadath — APPROVED** [review]
